pkg/resmap/resmap.go

```diff
@@ -552,26 +552,16 @@ func (m *resWrangler) makeCopy(copier resCopier) ResMap {
 // SubsetThatCouldBeReferencedByResource implements ResMap.
 func (m *resWrangler) SubsetThatCouldBeReferencedByResource(
 	inputRes *resource.Resource) ResMap {
-	inputId := inputRes.OrgId()
-	if !inputId.IsNamespaceableKind() {
-		if inputRes.GetOutermostNamePrefix() == "" {
-			return m
-		}
-		result := New()
-		for _, r := range m.Resources() {
-			if r.GetOutermostNamePrefix() == inputRes.GetOutermostNamePrefix() &&
-				r.GetOutermostNameSuffix() == inputRes.GetOutermostNameSuffix() {
-				err := result.Append(r)
-				if err != nil {
-					panic(err)
-				}
-			}
-		}
-		return result
-	}
 	result := New()
+	inputId := inputRes.CurId()
+	rctxm := inputRes.PrefixesSuffixesEquals
 	for _, r := range m.Resources() {
-		if !r.OrgId().IsNamespaceableKind() || inputRes.InSameFuzzyNamespace(r) {
+		// Need to match more accuratly both at the time of selection and transformation.
+		// OutmostPrefixSuffixEquals is not accurate enough since it is only using
+		// the outer most suffix and the last prefix. Use PrefixedSuffixesEquals instead.
+		if (!inputId.IsNamespaceableKind() && r.InSameKustomizeCtx(rctxm)) ||
```

Comment:
```
(!inputId.IsNamespaceableKind() || !r.OrgId().IsNamespaceableKind || r.CurId.IsNsEquals(input)) && r.InSameKustomizeCtx(rctxm)
```

Reply:
@Liujingfang1 Sorry. Should have seen that. This is done now. Ready for merge I hope.

Reply:
Yes, thank you.